### PR TITLE
Custom render passes on a camera

### PR DESCRIPTION
**Pull request by mvaligursky**
- an API (private for now) on the Camera to allow render passes for its rendering to be added to it.
- those passes are then inserted into the frame graph by forward renderer, allowing rendering the camera using custom render passes.

Suggestion: Hide whitespaces when reviewing.

### CODE REVIEW

File: src/scene/composition/render-action.js [hunk 1]
```diff
@@ -59,6 +62,12 @@ class RenderAction {
         this.viewBindGroups.length = 0;
     }
 
+    setupClears(camera, layer) {
+        this.clearColor = camera?.clearColorBuffer || layer.clearColorBuffer;
```
Comment (slimbuck) [inline src/scene/composition/render-action.js:66]: I assume `||` is correct here instead of `??`?

File: src/scene/frame-graph.js [hunk 2]
```diff
@@ -24,7 +24,28 @@ class FrameGraph {
      */
     addRenderPass(renderPass) {
         Debug.assert(renderPass);
-        this.renderPasses.push(renderPass);
+
+        const passes = this.renderPasses;
+
+        const beforePasses = renderPass.beforePasses;
```
Comment (slimbuck) [inline src/scene/frame-graph.js:30]: Why not?
`passes = renderPass.beforePasses.filter(p => p.enabled)`

**Review by slimbuck — APPROVED** [review]

File: src/scene/composition/render-action.js (hunk 1 above)
Comment (mvaligursky) [reply src/scene/composition/render-action.js:66]: both work here equally.

File: src/scene/frame-graph.js (hunk 2 above)
Comment (mvaligursky) [reply src/scene/frame-graph.js:30]: this runs every frame, so performance is the reason
https://plainenglish.io/blog/are-for-loops-better-than-arrays-filter-or-foreach-methods

**Pull request merged**